src/sentry/models/groupassignee.py

```diff
@@ -21,24 +21,34 @@
 
 class GroupAssigneeManager(BaseManager):
     def assign(self, group, assigned_to, acting_user=None):
+        from sentry.models import User, Team
+
+        if isinstance(assigned_to, User):
+            assignee_type = 'user'
+        elif isinstance(assigned_to, Team):
+            assignee_type = 'team'
+        else:
+            raise AssertionError('Invalid type to assign to: %r' % type(assigned_to))
+
         now = timezone.now()
         assignee, created = GroupAssignee.objects.get_or_create(
             group=group,
             defaults={
                 'project': group.project,
-                'user': assigned_to,
+                assignee_type: assigned_to,
                 'date_added': now,
             }
         )
 
         if not created:
             affected = GroupAssignee.objects.filter(
                 group=group,
-            ).exclude(
-                user=assigned_to,
-            ).update(
-                user=assigned_to, date_added=now
-            )
+            ).exclude(**{
+                assignee_type: assigned_to,
+            }).update(**{
+                assignee_type: assigned_to,
+                'date_added': now,
+            })
```

Comment:
Shouldn't this null out the other type field if it's updating the row (e.g. set user to `None` if you're setting team and vice versa.)

Reply:
Good call

Reply:
Fixed in https://github.com/getsentry/sentry/pull/7080/commits/e1bfec9a28ef4e3784c9c940c423ce54c12de7c9 and added tests.